### PR TITLE
configure: Detect MSVC Version Using C Macro

### PR DESCRIPTION
**Pull request by iguessthislldo**
Fix for #1110.

Instead of relying on using regex on the version message, which might fail in at least one major localization (#1110) or future versions (#922).

Also fixes the final message for VS users to tell them how they can open the sln file right then and there.

### CODE REVIEW

File: configure [hunk 1]
```diff
@@ -470,32 +470,57 @@ if ($opts{'compiler'} =~ /cl(\.exe)?$/i) {
...
+      "script from the Visual Studio Command Prompt.\nStopped";
+  my $ver = 0;
+  my $arch = '';
+  while (my $line = <$cl_out_fd>) {
+    chomp ($line);
+    print "CL says: $line\n" if $opts{'verbose'};
+    # Convert _MSC_VER to the form \d+\.\d
+    if ($line =~ /^CL is (\d+)\d (\w+)$/) {
+      $ver = int($1) / 10;
+      $arch = $2;
```
Comment (mitza-oci) [inline configure]: This `$arch` is passed down to cmake, so I don't think `x32` is valid.
https://cmake.org/cmake/help/v3.14/generator/Visual%20Studio%2016%202019.html

Comment (iguessthislldo) [reply configure]: Alright, didn't think of what it was being used for, but I can fix that. Should we worry about the ARM targets at all?

Comment (mitza-oci) [reply configure]: I wouldn't worry too much about the ARM targets at this point, but if we can just pass it through that should be fine.  Looks like the function `win32_cmake_arch` may need to change for this.

Comment (iguessthislldo) [reply configure]: > I wouldn't worry too much about the ARM targets at this point, but if we can just pass it through that should be fine.

Okay

> This $arch is passed down to cmake, so I don't think x32 is valid.

`win32_cmake_arch()` and `target_win64` both only look for `x64`, so actually `x32` works. Although, I'm not quite sure why I changed it from `x86` in your original comment to `x32`, so I think I will change that even if it works.

Comment (mitza-oci) [reply configure]: The ones in my original notes were from running the compiler and looking at its "logo" output:
```
Microsoft (R) C/C++ Optimizing Compiler Version 19.21.27702.2 for x86
```
VS uses `Win32` as the platform name and it looks like `cmake -A` follows this.  I was thinking that our `win32_cmake_arch` could be more future-proof.

**Comment by mitza-oci** [comment]
Doesn't look like it's valid Perl?
```
Can't modify constant item in scalar assignment at configure line 53, near "},"
```

**Comment by iguessthislldo** [comment]
Yea it's not, pushing a fix, I should've run it before commiting.

**Comment by iguessthislldo** [comment]
[Looks a qt server vcpkg was using went down](https://dev.azure.com/opendds/OpenDDS/_build/results?buildId=566&view=logs), Restarting the windows builds.

**Pull request merged**